.infrastructure/helm-chart/todoapp/charts/mysql/templates/secret.yml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Chart.Name }}-secrets
+  namespace: {{ .Values.mysql.namespace }}
+type: Opaque
+data:
+  {{- range $k, $v := .Values.mysql.secret }}
+    {{ $k }}: {{ $v }}
```

Comment:
Ensure that the values in `.Values.mysql.secret` are base64 encoded. Kubernetes requires secret data to be encoded in base64, so verify that the values provided in your `values.yaml` file are correctly encoded.